src/test/java/teammates/test/cases/logic/FeedbackSessionsLogicTest.java

```diff
@@ -96,6 +97,76 @@ public void testAll() throws Exception {
+            throws InvalidParametersException, EntityDoesNotExistException {
+        FeedbackResponseAttributes responseToBeDeleted = getResponseFromDatastore("response2ForQ2S2C1",
+                                                                                  dataBundle);
+        assertEquals(1, numResponsesFromGiverInSession(responseToBeDeleted.giverEmail,
+                                                       responseToBeDeleted.feedbackSessionName));
+
+        int originalResponseRate = getResponseRate(responseToBeDeleted.feedbackSessionName,
+                                                   responseToBeDeleted.courseId);
+
+        frLogic.deleteFeedbackResponseAndCascade(responseToBeDeleted);
+        fsLogic.updateSessionResponseRateForDeletedStudentResponse(responseToBeDeleted.giverEmail,
+                responseToBeDeleted.feedbackSessionName, responseToBeDeleted.courseId);
+        int responseRateAfterDeletion = getResponseRate(responseToBeDeleted.feedbackSessionName,
+                                                        responseToBeDeleted.courseId);
+
+        assertEquals(originalResponseRate - 1, responseRateAfterDeletion);
+
+        restoreStudentFeedbackResponseToDatastore(responseToBeDeleted);
+    }
+
+    private void updateResponseRate_hasRemainingResponsesFromStudent_noChangeToResponseRate()
+            throws InvalidParametersException, EntityDoesNotExistException {
+        FeedbackResponseAttributes responseToBeDeleted = getResponseFromDatastore("response1ForQ1S1C1",
+                                                                                  dataBundle);
+        assertTrue(1 < numResponsesFromGiverInSession(responseToBeDeleted.giverEmail,
+                                                      responseToBeDeleted.feedbackSessionName));
+
+        int originalResponseRate = getResponseRate(responseToBeDeleted.feedbackSessionName,
+                                                   responseToBeDeleted.courseId);
+
+        frLogic.deleteFeedbackResponseAndCascade(responseToBeDeleted);
+        fsLogic.updateSessionResponseRateForDeletedStudentResponse(responseToBeDeleted.giverEmail,
+                responseToBeDeleted.feedbackSessionName, responseToBeDeleted.courseId);
+        int responseRateAfterDeletion = getResponseRate(responseToBeDeleted.feedbackSessionName,
+                                                        responseToBeDeleted.courseId);
+
+        assertEquals(originalResponseRate, responseRateAfterDeletion);
+
+        restoreStudentFeedbackResponseToDatastore(responseToBeDeleted);
+    }
+
+    private int numResponsesFromGiverInSession(String studentEmail, String sessionName) {
+        int numResponses = 0;
+        for (FeedbackResponseAttributes response : dataBundle.feedbackResponses.values()) {
+            if (response.giverEmail.equals(studentEmail) && response.feedbackSessionName.equals(sessionName)) {
```

Comment:
You should check for course here too, since feedback session names are only unique to a course.